plutus-core/plutus-core/src/PlutusCore/Parser/ParserCommon.hs

```diff
@@ -114,7 +114,19 @@ inBraces :: Parser a -> Parser a
 inBraces = between (symbol "{") (char '}')
 
 isIdentifierChar :: Char -> Bool
-isIdentifierChar c = isAlphaNum c || c == '_' || c == '\''
+isIdentifierChar c = isAlphaNum c || elem c identifierSymbols
```

Comment:
I could complain that this should be called `isIdentifierNonStartingChar` or something, but that would be horrible.